### PR TITLE
Forbid most side effects in select labels

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -5228,9 +5228,11 @@ verify(false, error.NoMatch);
...
 expression be compile-time known values. Keysets are evaluated in
 order, from top to bottom as implied by the pseudo-code above; the
-first keyset that includes the value in the `select` argument
-provides the result state. If no label matches, the execution triggers
-a runtime error with the standard error code `error.NoMatch`.
+first keyset that includes the value in the `select` argument provides
+the result state. If no label matches, the execution triggers a
+runtime error with the standard error code `error.NoMatch`. Function
+or method calls are forbidden in keyset expressions, except calls to
+the header methods `isValid()`, `minSizeInBits()`, and `minSizeInBytes()`.
```
Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:5235]: The first sentence of the paragraph says "Some targets may require that all keyset expression in a select expression be compile-time known values."

Are you proposing here that even for implementations that allow keyset expressions that are not compile-time known values, the only function or method calls permitted are ones that call the method `isValid()`?

Note that `minSizeInBits()` and `minSizeInBytes()` are already specified to return compile-time known values.

Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk:5235]: yes. no target allows such calls.

Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk:5235]: Do any targets support keyset expressions whose values are not compile-time known values?  (I believe that the name of a parser value set object instance is a compile-time known value.)

Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk:5235]: Is this a breaking change, which changes the set of accepted programs?

Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk:5235]: We should think about how this can be implemented in the compiler with side-effects.

**Comment by mbudiu-vmw** [comment]
The LDWG has agreed that this should not be accepted.

**Pull request closed without merging**